azure-mgmt-network/src/main/java/com/microsoft/azure/management/network/implementation/NicIpConfigurationImpl.java

```diff
@@ -237,16 +239,22 @@ public NicIpConfigurationImpl withSubnet(String name) {
     }
 
     @Override
-    public NicIpConfigurationImpl withExistingLoadBalancer(LoadBalancer loadBalancer) {
-        this.loadBalancerToAssociate = loadBalancer;
-        return this;
+    public NicIpConfigurationImpl withExistingLoadBalancerBackend(LoadBalancer loadBalancer, String backendName) {
+        for (BackendAddressPoolInner pool : loadBalancer.inner().backendAddressPools()) {
+            if (pool.name().equalsIgnoreCase(backendName)) {
+                ensureBackendAddressPools().add(pool);
+                return this;
+            }
+        }
+
+        return null;
```

Comment:
...so to make their error checking robust, the user code should fist ensure that the backend exists:
```java
if (!lb.backends().contains(backendName)) {
    // Missing backend
} else {
    // Call the code in question
}
```